src/ripple/ledger/tests/View_test.cpp

```diff
@@ -524,6 +525,209 @@ class View_test
     }
 
     void
+    testFlags()
+    {
+        using namespace jtx;
+        Env env(*this);
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const carol = Account("carol");
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        auto const EUR = gw["EUR"];
+
+        env.fund(XRP(10000), alice, bob, carol, gw);
+        env.trust(USD(100), alice, bob, carol);
+        {
+            // Global freezing.
+            env(pay(gw, alice, USD(50)));
+            env(offer(alice, XRP(5), USD(5)));
+
+            // Now freeze gw.
+            env(fset (gw, asfGlobalFreeze));
+            env.close();
+            env(offer(alice, XRP(4), USD(5)), ter(tecFROZEN));
+            env.close();
+
+            // Alice's USD balance should be zero if frozen.
+            expect(USD(0) == accountHolds (*env.closed(),
+                alice, USD.currency, gw, fhZERO_IF_FROZEN, env.journal));
+
+            // Thaw gw and try again.
+            env(fclear (gw, asfGlobalFreeze));
+            env.close();
+            env(offer("alice", XRP(4), USD(5)));
+        }
+        {
+            // Local freezing.
+            env(pay(gw, bob, USD(50)));
+            env.close();
+
+            // Now gw freezes bob's USD trust line.
+            env(trust(gw, USD(100), bob, tfSetFreeze));
+            env.close();
+
+            // Bob's balance should be zero if frozen.
+            expect(USD(0) == accountHolds (*env.closed(),
+                bob, USD.currency, gw, fhZERO_IF_FROZEN, env.journal));
```

Comment:
See my comment on global freeze. We should be testing the all four combinations for Local freeze and freeze handling.